### PR TITLE
web-wallet: Split transfer contract UI

### LINKED ISSUE #2175: Split Transfer contract UI into 2 routes
The contract UI needs to be split into 2 separate routes "send" and "receive.

Figma file:
https://www.figma.com/design/z6UnA0BtMBUKIcCXD975um/Wallet-Extension-(JULY-26th)?node-id=6207-159257&m=dev

### PR DESCRIPTION
**Pull request by kieranhall**
Resolves #2175

### CODE REVIEW

File: web-wallet/src/routes/(app)/dashboard/receive/+page.svelte [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  import { contractDescriptors, updateOperation } from "$lib/contracts";
+  import { operationsStore } from "$lib/stores";
+  import { mdiArrowTopRight } from "@mdi/js";
+
+  onMount(() => {
+    updateOperation("receive");
+  })
+
+  onDestroy(() => {
+    updateOperation("receive");
```
Comment (ascartabelli) [inline web-wallet/src/routes/(app)/dashboard/receive/+page.svelte:16]: Why "receive" again?

File: web-wallet/src/routes/(app)/dashboard/receive/+page.svelte [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+</script>
+
+<IconHeadingCard
+  gap="medium"
+  heading="Transfer"
+  iconPath={mdiArrowTopRight}
+  reverse
+>
+  <TransferContract
+    descriptor={contractDescriptors[1]}
```
Comment (ascartabelli) [inline web-wallet/src/routes/(app)/dashboard/receive/+page.svelte:27]: We should have a function to retrieve them by ID.

File: web-wallet/src/routes/(app)/dashboard/receive/+page.svelte [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    on:operationChange={({ detail }) => updateOperation(detail)}
```
Comment (ascartabelli) [inline web-wallet/src/routes/(app)/dashboard/receive/+page.svelte:28]: Should we rethink the `TransferContract`?
It's a bit confusing having this page update the operation value, then leave the freedom to the inner component to do the same.

File: web-wallet/src/routes/(app)/dashboard/staking/+page.svelte [hunk 4]
```diff
@@ -32,7 +21,7 @@
     reverse
   >
     <StakeContract
-      descriptor={contractDescriptors[1]}
+      descriptor={contractDescriptors[2]}
       on:operationChange={({ detail }) => updateOperation(detail)}
       on:suppressStakingNotice={() => updateSetting("hideStakingNotice", true)}
```
Comment (ascartabelli) [inline web-wallet/src/routes/(app)/dashboard/staking/+page.svelte:26]: You removed the function that was called here.

File: web-wallet/src/routes/(app)/dashboard/send/+page.svelte
Comment (ascartabelli) [inline web-wallet/src/routes/(app)/dashboard/send/+page.svelte:1]: Same comments made about the receive page.

File: web-wallet/src/routes/(app)/dashboard/receive/+page.svelte (hunk 3 above)
Comment (ascartabelli) [reply web-wallet/src/routes/(app)/dashboard/receive/+page.svelte:28]: On second thought: as you're separating things: can we get rid of that "operation store" entirely?

File: web-wallet/src/routes/(app)/dashboard/receive/+page.svelte (hunk 1 above)
Comment (kieranhall) [reply web-wallet/src/routes/(app)/dashboard/receive/+page.svelte:16]: This is the operation for receiving tokens.

File: web-wallet/src/routes/(app)/dashboard/receive/+page.svelte (hunk 2 above)
Comment (kieranhall) [reply web-wallet/src/routes/(app)/dashboard/receive/+page.svelte:27]: I think we should remove the contract descriptors.

File: web-wallet/src/routes/(app)/dashboard/receive/+page.svelte (hunk 3 above)
Comment (kieranhall) [reply web-wallet/src/routes/(app)/dashboard/receive/+page.svelte:28]: Basically, yes.  But perhaps in it's own PR?

File: web-wallet/src/routes/(app)/dashboard/staking/+page.svelte (hunk 4 above)
Comment (kieranhall) [reply web-wallet/src/routes/(app)/dashboard/staking/+page.svelte:26]: I don't follow.  I just updated the contractDescriptors.  Show me tomorrow what you are talking about here.

**Review by ascartabelli — DISMISSED** [review]

**Pull request merged**